### ISSUE TITLE
Institution code used incorrectly

### ISSUE THREAD

**Issue by gbif-portal** [gbif/portal-feedback#2152]
**Institution code used incorrectly**

It appears that all 24K+ records in the dataset from the Herbarium collecton of the National University of La Plata use a URL-like string instead of their assigned Institution Code of '[LPC](`https://www.gbif.org/grscicoll/institution/afd6368e-17cb-430a-ba65-9b0aade9f3e9)'.` Hope it might be an easy bulk fix at some point.

-----

Github user: @kcopas
User: [See in registry](https://www.gbif.org/api/feedback/user/0459bfde9eded34151c03d5a03965b5d:1b30febd207d54a0a7c03c85a21148b4d466bd7b0d1b51a37f7f0e2cb8ba332aaf4976209d4def6b6ae548b5415985a922dbfe2ed114fa74db6b6d09b6f4ffb0)
System: Firefox 68.0.0 / Mac OS X 10.14.0
Referer: https://www.gbif.org/dataset/b4a37621-7556-4705-9349-ede261feebe4
Window size: width 1654 - height 931
[API log](http://elk.gbif.org:5601/app/kibana?#/discover?_g=(refreshInterval:(display:Off,pause:!f,value:0),time:(from:'2019-08-09T15:15:44.452Z',mode:absolute,to:'2019-08-09T15:21:44.452Z'))&_a=(columns:!(_source),index:'prod-varnish-*',interval:auto,query:(query_string:(analyze_wildcard:!t,query:'response:%3E499')),sort:!('@timestamp',desc)))
[Site log](http://elk.gbif.org:5601/app/kibana?#/discover?_g=(refreshInterval:(display:Off,pause:!f,value:0),time:(from:'2019-08-09T15:15:44.452Z',mode:absolute,to:'2019-08-09T15:21:44.452Z'))&_a=(columns:!(_source),index:'prod-portal-*',interval:auto,query:(query_string:(analyze_wildcard:!t,query:'response:%3E499')),sort:!('@timestamp',desc)))
System health at time of feedback: OPERATIONAL

**Comment by ManonGros** [comment]
Is having a URL instead of a standard name or code an issue? I have seen other datasets with URL for institutions as well (for example [this one](https://www.gbif.org/dataset/f2654dba-a152-4731-807c-4c90b75ced00)). I don't really see the problem.
https://dwc.tdwg.org/terms/#dwc:institutionCode

**Comment by kcopas** [comment]
It doesn't really seem in line the intent of the term, imho. Wouldn't a URL be better placed elsewhere?

The institution code is also present in GRSciColl. It seems a logical term to use in crosslinking, rather than using it as a catch-all.

**Comment by ManonGros** [comment]
Aren't URLs more precise than these Codes? The codes might change and the two institution might have the same one. GrSciColl has identifiers (e.g. URLs) which resolve to the institution page and I have seen several publishers use that.
Wouldn't it make more sense than using codes?

**Comment by MattBlissett** [comment]
The institutions codes coming from Index Herbariorum have existed for over 50 years, and are unique (within that list of codes).

https://en.wikipedia.org/wiki/Index_Herbariorum

**Comment by ManonGros** [comment]
Fair enough but what about the other collection? There is no uniqueness constraint in GrSciColl.
Plus https://dwc.tdwg.org/terms/#dwc:institutionCode doesn't specify the source of the code.

**Comment by MortenHofft** [comment]
**When publishing**
I would think that institutionId was a better choice for URLs and save codes a la `NHM` for institutionCode. But it seems reasonable that we could handle either. I wonder if there is a practice of qualifying it with a prefix `IH.CAL`

**In GrSciColl**
And perhaps extend GrSciColl to support multiple codes/synonyms/deprecatedCodes/combinedWith (perhaps with qualifiers? _In IH the code is CAL_). 

**In our processing**
And we could start to actually interpret these fields - similar to what we do with taxonomy - and try to match the incoming data to unique GrSciColl identifiers (whatever those are). And in that process we would accept URLs in institutionCodes (perhaps adding a flag indicating that it isn't recommended usage).

**Comment by jhnwllr** [comment]
Marking as pending to be handled de-centrally at some point.
